### PR TITLE
Mylin/#234 concatenate stoke images

### PR DESCRIPTION
**Pull request by acdo2002**
Close [Issue#234](https://github.com/CARTAvis/carta-backend-ICD-test/issues/234).
This is the test to check concatenate ICD message: several stoke images can be merged into one single image.

### CODE REVIEW

**Review by ajm-asiaa — APPROVED** [review]
Seems to work well.

File: src/test/CONCAT_STOKES_IMAGES.test.ts [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+let concatStokeTimeout = config.timeout.concatStokes;
+
+interface ConcatStokesFilesAckExt extends CARTA.IConcatStokesFilesAck {
+    OpenFileAckBeamLength: number
+}
+
+interface AssertItem {
+    register: CARTA.IRegisterViewer;
+    filelist: CARTA.IFileListRequest;
+    fileOpen: CARTA.IOpenFile;
```
Comment (zarda) [inline src/test/CONCAT_STOKES_IMAGES.test.ts:19]: May need to remove

File: src/test/CONCAT_STOKES_IMAGES.test.ts [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+        test(`(Step 2) Modify assert concatenate directory and request CONCAT_STOKES_FILES_ACK within ${concatStokeTimeout} ms | `,async()=>{
+            assertItem.ConcatReq.stokesFiles.map((input,index)=>{
+                assertItem.ConcatReq.stokesFiles[index].directory = basePath + `/` + testSubdirectory; //`${basePath}/` + testSubdirectory;
+            });
+
+            await Connection.send(CARTA.CloseFile, { fileId: -1 });
+            await Connection.send(CARTA.ConcatStokesFiles,assertItem.ConcatReq);
+            await Connection.receive(CARTA.RegionHistogramData);
+            ConcatStokesResponse = await Connection.receive(CARTA.ConcatStokesFilesAck);
+            // console.log(ConcatStokesResponse);
```
Comment (zarda) [inline src/test/CONCAT_STOKES_IMAGES.test.ts:136]: Better to clean the code for debug

**Review by zarda — COMMENTED** [review]
Overall is tested well and ready to merge.

**Comment by acdo2002** [comment]
@zarda I have improved the codes based on your suggestions.
@ajm-asiaa there are new added test: CONCAT_STOKES_IMAGES_AXIS_DEGENERACY.test.ts & CONCAT_ERROR_MESSAGE.test.ts, could you verify them again?

The updated test document is [here](https://docs.google.com/document/d/1gjbuDzWQ_6TExipxp6HT0ZhlRbdm2Ba3U0JRmx_5gyc/edit#)

**Review by zarda — APPROVED** [review]

**Pull request merged**